### DISCUSSION TITLE
Problem installing into IRIS for Health 2022.1.0.209.0

### DISCUSSION THREAD

**Discussion by cschaabuphp** [intersystems/git-source-control#203] [General]
I am trying to install git-source-control and I am getting:
=============================================================================
|| Welcome to the Package Manager Shell (ZPM).                             ||
|| Enter q/quit to exit the shell. Enter ?/help to view available commands ||
=============================================================================
zpm:GIT-TEST>install git-source-control -v
 
Module git-source-control was downloaded from registry (https://pm.community.intersystems.com)
Loading from C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\
 
Load started on 08/11/2022 13:56:11
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\module.xml as xml
Imported document: git-source-control.ZPM
Load finished successfully.
 
Skipping preload - directory does not exist.
Load started on 08/11/2022 13:56:11
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\module.xml as xml
Imported document: git-source-control.ZPM
Load finished successfully.
 
Loading git-source-control in process 14364
[git-source-control]    Reload START (C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\)
Skipping preload - directory does not exist.
Load of directory started on 08/11/2022 13:56:11 '*'
 
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\API.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\Change.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\Extension.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\Modification.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\PackageManagerContext.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\PullEventHandler.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\Settings.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\StreamServer.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\Utils.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\WebUIDriver.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\PullEventHandler\Default.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\PullEventHandler\IncrementalLoad.cls as udl
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\cls\SourceControl\Git\PullEventHandler\PackageManager.cls as udl
Load finished successfully.
 
Load started on 08/11/2022 13:56:13
Loading file C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\inc\SourceControl\Git.inc as udl
SourceControl.Git.INC Loaded
Load finished successfully.
 
[git-source-control]    Reload SUCCESS
[git-source-control]    Module object refreshed.
[git-source-control]    Validate START
[git-source-control]    Validate SUCCESS
[git-source-control]    Compile START
Compilation started on 08/11/2022 13:56:13 with qualifiers 'd-lck'
Compiling 13 classes
Compiling class SourceControl.Git.API
Compiling class SourceControl.Git.Change
Compiling class SourceControl.Git.Extension
Compiling class SourceControl.Git.Modification
Compiling class SourceControl.Git.PackageManagerContext
Compiling class SourceControl.Git.PullEventHandler
Compiling class SourceControl.Git.Settings
Compiling class SourceControl.Git.StreamServer
Compiling class SourceControl.Git.Utils
Compiling class SourceControl.Git.WebUIDriver
Compiling class SourceControl.Git.PullEventHandler.IncrementalLoad
Compiling class SourceControl.Git.PullEventHandler.PackageManager
Compiling class SourceControl.Git.PullEventHandler.Default
Compiling table SourceControl_Git.Change
Compiling routine SourceControl.Git.API.1
Compiling routine SourceControl.Git.Change.1
Compiling routine SourceControl.Git.Extension.1
Compiling routine SourceControl.Git.Modification.1
Compiling routine SourceControl.Git.PackageManagerContext.1
Compiling routine SourceControl.Git.PullEventHandler.1
Compiling routine SourceControl.Git.Settings.1
Compiling routine SourceControl.Git.StreamServer.1
Compiling routine SourceControl.Git.Utils.1
Compiling routine SourceControl.Git.WebUIDriver.1
Compiling routine SourceControl.Git.PullEventHandler.IncrementalLoad.1
Compiling routine SourceControl.Git.PullEventHandler.PackageManager.1
Compiling routine SourceControl.Git.PullEventHandler.Default.1
Compilation finished successfully in 0.753s.
 
[git-source-control]    Compile SUCCESS
[git-source-control]    Activate START
Copying C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\csp\gitprojectsettings.csp to C:\InterSystems\IRISHealth-Dev\devuser\studio\templates\gitsourcecontrol\gitprojectsettings.csp as file
Copying C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\csp\webuidriver.csp to C:\InterSystems\IRISHealth-Dev\devuser\studio\templates\gitsourcecontrol\webuidriver.csp as file
Copying C:\InterSystems\IRISHealth-Dev\Mgr\.modules\GIT-TEST\git-source-control\2.0.2\git-webui\release\share\git-webui\webui\ to C:\InterSystems\IRISHealth-Dev\devuser\studio\templates\gitsourcecontrol\ as directory
[git-source-control]    Activate FAILURE
ERROR! ObjectScript error: <NOTOPEN>zRunCommandViaZF+14^%ZPM.PackageManager.Developer.Utils.1
zpm:GIT-TEST>

Any help would be appreciated,

Charlie

**Comment by isc-tleavitt** [comment]
What's your package manager version?
zpm "version"

**Reply by cschaabuphp** [reply]
zpm:GIT-TEST>version
 
%SYS> zpm 0.3.4
 Locally installed zpm-registry not found
https://pm.community.intersystems.com - 1.0.6
zpm:GIT-TEST>

**Reply by isc-tleavitt** [reply]
@cschaabuphp looking at the package manager code, the line where the failure occurs is a $zf(-1) call. (It's using ROBOCOPY on Windows to do the file copy.)

On some level, this is an issue with the package manager itself. You presumably don't have the %System_Callout:USE privilege, but you shouldn't need it to just be able to copy files from point A to point B. You will, however, need %System_Callout:USE for the extension to work, because it uses $zf(-100) to issue git commands.

See for reference: https://docs.intersystems.com/iris20221/csp/docbook/DocBook.UI.Page.cls?KEY=BGCL_syscall and https://docs.intersystems.com/iris20221/csp/docbook/DocBook.UI.Page.cls?KEY=BGCL_syscall#BGCL_syscall_privilege

I've reported two issues against zpm stemming from this investigation:
https://github.com/intersystems-community/zpm/issues/336
https://github.com/intersystems-community/zpm/issues/337

If you do in fact have %System_Callout:USE, there could be something else going on.

**Reply by isc-tleavitt** [reply]
Assuming I've diagnosed the root cause correctly, also filed https://github.com/intersystems/git-source-control/issues/204 which would prevent confusion the next time someone runs into this.

**Reply by cschaabuphp** [reply]
@isc-tleavitt I appreciate your help. I have the roll %All. So I assume that has %System_Callout:USE. I am pretty new to Intersystems so I don't know how to verify that. Could it be a Windows permission issue? Is there a way to see exactly what it is trying to execute when it fails? We recently changed the user the Intersystems service is running under. That user is a local admin.

**Reply by isc-tleavitt** [reply]
@cschaabuphp from terminal you could:
w $System.Security.Check("%System_Callout","USE")

If that returns 1 (or if not) I'd be happy to connect via a Teams call or something to debug. Other steps would be looking in the audit database and/or getting more information about the error that's actually happening.

**Reply by cschaabuphp** [reply]
It returned 1 and I am out of ideas. I would appreciate a meeting. How do we do that?

**Reply by isc-tleavitt** [reply]
I'll reach out via email

**Reply by isc-tleavitt** [reply]
Root cause appears to be PATH setup for the user the InterSystems service is running under. Further debugging steps after updating the path would be to review the audit database to see which commands are running and getting <NOTOPEN>.

**Reply by cschaabuphp** [reply]
Actually it wasn't the PATH. It was because when I changed the user the service runs as. I just changed the service config in windows. I didn't run <install-dir>\bin\IRISinstall.exe setserviceusername <instance-name> <username> <password>. So I ran that and it now works. Thanks for you help Tim.
